### DISCUSSION TITLE
Error querying the database: Protocol error: invalid token type 0

### DISCUSSION THREAD

**Discussion by nappalm** [prisma/prisma#24635] [Q&A]
### Question

Hello everyone,
I am trying to connect to a sql server 2000 database, I have followed the prisma procedure for its configuration, however I have created an endpoint to check its status using express getting the following error:

`Error querying the database: Protocol error: invalid token type 0
`
The connection data is correct, I have connected without problems using Dbeaver.

---

Connection data in .env file
`DATABASE_URL='sqlserver://192.168.1.1:14303;initial catalog=CtrlChks;database=CtrlChks;user=Ctrlchq_dev;password={L46Ds"srD=V<8*};encrypt=DANGER_PLAINTEXT;connect_timeout=30;sslmode=prefer'`

--- 

app.js

```
const express = require('express');
const { PrismaClient, Prisma } = require('@prisma/client');

const app = express();
const prisma = new PrismaClient({
  log: ['query', 'info', 'warn', 'error'],
});

 app.get('/status', async (req, res) => {
  try {
    await prisma.$connect();
    res.send('Database connection successful');
  } catch (error) {
    //🔴  Error querying the database: Protocol error: invalid token type 0
    res.status(500).send(`Database connection failed: ${error.message}`);
  } finally {
    await prisma.$disconnect();
  }
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Server is running on port ${PORT}`);
});
```
![image](https://github.com/prisma/prisma/assets/35016608/61b85b90-ca3b-4897-a342-c765578d8904)

Can someone help me, if it is a problem of sql server 2000 or I need to configure something in prisma.
I have created only one model in schema.prisma to test then I have run npx prisma generate.
I have also run npx prisma db pull, getting the same error:
`Error: Error querying the database: Protocol error: invalid token type 0`

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model EstatusFact {
  IdEstFac String @id @db.Char(255) 
  Descrip  String @db.Char(255)
}

```

- OS: Windows
- Database: SQL Server
- Node.js version: 20.11.1

 prisma                  : 5.16.0
@prisma/client          : 5.15.1
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64

**Comment by nappalm** [comment]
I had the same problem with sequelize, however in sequelize it worked for me using `tdsVersion: "7_1"` inside options, in prisma how can I replace that?

**Comment by ludralph** [comment]
Hi @nappalm 👋 

SQL Server 2000 is not supported by Prisma. Prisma has support for SQL Server 2017, 2019 and 2022. You can take a look at the list of supported databases [here](https://www.prisma.io/docs/orm/reference/supported-databases).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
